nameserver/nameserver.go

```diff
@@ -161,8 +161,9 @@ func (n *Nameserver) PeerGone(peer *router.Peer) {
 func (n *Nameserver) Delete(hostname, containerid, ipStr string, ip address.Address) error {
 	n.Lock()
 	n.infof("tombstoning hostname=%s, container=%s, ip=%s", hostname, containerid, ipStr)
+	lowerHostname := strings.ToLower(hostname)
 	entries := n.entries.tombstone(n.ourName, func(e *Entry) bool {
-		if hostname != "*" && e.Hostname != hostname {
+		if hostname != "*" && strings.ToLower(e.Hostname) != lowerHostname {
```

Comment:
why do we need the `ToLower` here? It's an entry hostname, so surely it should be lower-case already.